src/shared/components/CreditCardForm/index.tsx

```diff
@@ -1,20 +1,32 @@
-import React, {FC, useContext, useEffect, useState} from 'react'
+import React, {FC, useEffect, useRef, useState} from 'react'
 
 // Context
-import {CheckoutContext} from 'src/checkout/context/checkout'
+import {CreditCardParams} from 'src/client/unityRoutes'
+import {ZuoraClient} from 'src/types/billing'
 
 export const ZUORA_SCRIPT_URL =
   'https://apisandboxstatic.zuora.com/Resources/libs/hosted/1.3.0/zuora-min.js'
 export const ZUORA_ID = 'zuora_payment'
 
-const ZuoraPaymentForm: FC = () => {
-  // FIXME: Add onFocus functionality
-  const {zuoraParams, handleSubmit, inputs} = useContext(CheckoutContext)
+export interface Props {
+  zuoraParams: CreditCardParams
+  onSubmit?: (paymentMethodId) => void
+  zuoraClient?: ZuoraClient
+}
 
-  const [client, setClient] = useState(window.Z)
+// FIXME: Add onFocus functionality
+const CreditCardForm: FC<Props> = ({zuoraParams, onSubmit, zuoraClient}) => {
+  const _isMounted = useRef(true)
```

Comment:
🤔 I'm a little fuzzy on `useRef` so my apologies for asking dumb questions - does `current` get set here on initialization? Also, what's the purpose of setting `isMounted` with a useRef here? I don't see it being set on a DOM node

Reply:
Yea, so the tests revealed that the request were made even after the form has been unmounted. This was causing memory leak as well as forcing tests to fail. The `current` for `_isMounted` gets set as `true` and when the component gets unmounted, it's set to `false`.

Reply:
Gotcha, I'm just trying to think of whether there's a clearer way of doing this that doesn't make use of `useRef` since we might find ourselves in a situation where we would need to list to changes made on the DOM. We could probably achieve the same thing by simply using `useState` and initializing a value in `useEffect`, etc... 

It should do the same thing, but I'm wary of `useRef` because of the way it propagates events. According to the [docs](https://reactjs.org/docs/hooks-reference.html#useref):

`Keep in mind that useRef doesn’t notify you when its content changes. `